### PR TITLE
chore: update docs for dfx canister

### PR DESCRIPTION
**Pull request by dfx-json**
# Description

Update documentation so that [internet](https://internetcomputer.org/docs/current/references/cli-reference/dfx-canister) contains accurate information

# How Has This Been Tested?

Used the output of `dfx --help` for each subcommand to cross reference against markdown reference

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: docs/cli-reference/dfx-canister.md [hunk 1]
```diff
@@ -766,3 +763,94 @@ To stop all of the canisters you have deployed on the `ic` {platform}, you can r
 ``` bash
 dfx canister --network=ic stop --all
 ```
+
+## dfx canister uninstall-code
+
+Use the `dfx canister stop` command to stop a canister that is currently running on the {platform} or on the local canister execution environment.
```
[the remark this answers is not in the document]
Comment (dfx-json) [reply docs/cli-reference/dfx-canister.md:769]: yes. good eye. it's easy for me to get lost in markdown apparently.

File: docs/cli-reference/dfx-canister.md [hunk 2]
```diff
@@ -766,3 +763,94 @@ To stop all of the canisters you have deployed on the `ic` {platform}, you can r
 ``` bash
 dfx canister --network=ic stop --all
 ```
+
+## dfx canister uninstall-code
+
+Use the `dfx canister uninstall-code` command to uninstall a canister that is currently running on the {platform} or on the local canister execution environment.
```
[the remark this answers is not in the document]
Comment (ericswanson-dfinity) [reply docs/cli-reference/dfx-canister.md:769]: Here is the [relevant comment](https://github.com/dfinity/agent-rs/blob/main/ic-utils/src/interfaces/management_canister.rs#L265) from agent-rs:

> This method removes a canister’s code and state, making the canister empty again. Only the controller of the canister can uninstall code. Uninstalling a canister’s code will reject all calls that the canister has not yet responded to, and drop the canister’s code and state. Outstanding responses to the canister will not be processed, even if they arrive after code has been installed again. The canister is now empty. In particular, any incoming or queued calls will be rejected.

[the remark this answers is not in the document]
Comment (dfx-json) [reply docs/cli-reference/dfx-canister.md]: It seems helpful to add that entire comment in the docs!

**Review by ericswanson-dfinity — APPROVED** [review]

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -7,6 +7,11 @@
 
 == DFX
 
+=== fix: Freezing Threshold now documented
+
+Calls made to retrieve the help output for `canister update-settings` was missing the `freezing-threshold` parameter.
+This is now included when running `dfx canister update-settings -h`.
+
```
Comment (ericswanson-dfinity) [inline CHANGELOG.adoc:14]: This actually needs to go in the "UNRELEASED" section (with a new `== DFX` header section)

**Pull request merged**